Feature/ppm 30 boardfarm docker compose

odkq opened the pull request:
This branch wraps the boardfarm tests using a python 3.5 script, dctest.py, that runs docker-compose.

The script allows for setting up the bridge with the physical device and cleaning up testing images seamlessly.

As the script is not running inside docker itself, it only uses Python 3.5 (introduced in 2015) features and no external libraries.

odkq committed:
PPM-30: Changes on class ALEntityDocker for compose

Use directly the hostname instead of querying for the ip

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
PPM-30: Define test entrypoint and json config

Entrypoint calls the json for "in compose" tests

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
PPM-30: Dockerfile and docker-compose.yml

Dockerfile to create boardfarm docker image and docker-compose.yml with
the configuration for boardfarm, controller and extender

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
PPM-30: dctest.py script to wrap docker-compose

It wraps docker-compose tests

- Verifies versions of docker and docker-compose
- Creates and launches boardfarm image with the correct environment
  variables
- Cleans images after building
- Returns value from boardfarm process inside the image

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
PPM-30: Add test to boardfarm-ci

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
PPM-30: Check return error on _run_shell

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
Added device prplmesh_compose.py

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
Use original name and docker-name for hostname in environment.py

Signed-off-by: odkq <pablo@odkq.com>
Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

rmelotte committed:
boardfarm: use the name of the container for the status check

The status are failing in CI.

The name of the container was hardcoded by mistake, use docker_name
instead.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+def check_docker_versions():
+    DOCKER_MAJOR = 19
+    DC_MAJOR = 1
+    DC_MINOR = 25
```
I already forgot why we require this specific version of docker (or higher).
Could you add a comment explaining why you chose this version?
Or is it because it's the version you tested it with? (nitpick)

odkq replied:
Probably any version > 18 will work with a modern docker-compose. As docker-compose changes more often the syntax of the docker-compose.yml file, this is to make sure a new version is installed.

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+        self.logdir = os.path.join(self.scriptdir, 'logs')
+        device_name = 'dockerized_device-{}'.format(self.build_id)
+        self.devicedir = os.path.join(self.logdir, device_name)
+        repeater_name = 'repeater1-{}'.format(self.build_id)
```
The name of the device have changed since then.
Is it still needed to create the folders in advance? if it is then we should either get the device name differently, or just use another approach to solve the file permissions problem.

odkq replied:
Indeed. As the .json file can change adding/moving devices I made a change to scan the .json file and create a directory per each of the devices present 0e1458da

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            print('Making {}'.format(self.repeaterdir))
+            os.makedirs(self.repeaterdir)
+
+    def cleanlogs(self):
```
This seems to be unused, did you forget to call it somewhere?

odkq replied:
Fixed on 61cc3b7

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+    def cleanlogs(self):
+        shutil.rmtree(os.path.join(self.scriptdir, 'logs'))
+
+    def _setNonBlocking(fd):
```
This doesn't seem to be used either. What is it there for?

aciliketcap replied:
Also _setNonBlocking should take "self" as first parameter if it is ever going to be used.

odkq replied:
Fixed on 61cc3b7

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            new_id = 1
+        else:
+            new_id = last_id + 1
+        return str(new_id)
```
If it's running outside CI, I think you can just use the username as an ID, and always re-use the same containers.
Having different container and names on each invocation is not a problem per se, except maybe if you want to run them quickly and skip the init (like it's done here: https://github.com/prplfoundation/prplMesh/pull/1589)

(nitpick)

odkq replied:
It is as useful locally as in CI for debugging, as logs are kept in separate directories for each run, thats why an 'id' is generated even if no parallel builds happen in the local machine.

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            new_id = last_id + 1
+        return str(new_id)
+
+    def copy_build_dir(self):
```
Isn't this also unused?

odkq replied:
Fixed in a0d8fe7b

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+    def dc(self, args, interactive=False):
+        params = ['docker-compose', '-f',
+                  'tools/docker/boardfarm-ci/docker-compose.yml']
+        # params += ['-p', 'boardfarm-ci-{}'.format(self.build_id)]
```
```suggestion
```

odkq replied:
Fixed in 727d8360

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+        local_env['ROOT_DIR'] = self.rootdir
+        docker_gid = grp.getgrnam('docker')[2]
+        local_env['CURRENT_UID'] = str(os.getuid()) + ':' + str(docker_gid)
+        local_env['CURRENT_ID'] = str(os.getuid())
```
The naming is a little bit confusing: CURRENT_ID is actually a UID, and CURRENT_UID is the combination of the UID and the GID
```suggestion
        local_env['CURRENT_UID_GID'] = str(os.getuid()) + ':' + str(docker_gid)
        local_env['CURRENT_UID'] = str(os.getuid())
```
(nitpick)

odkq replied:
Fixed on a0ae26cc

CURRENT_ID was passed to the controller and agent images when testing for running it as a regular user. Removed and renamed the CURRENT_UID as suggested

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            local_env['FINAL_ROOT_DIR'] = self.rootdir
+        else:
+            # Running inside gitlab-ci
+            local_env['FINAL_ROOT_DIR'] = '/builds/prpl-foundation/prplMesh'
```
I'm commenting for other reviewers (as well as my future self), this is needed until https://jira.prplfoundation.org/browse/PPM-208 is fixed.

Maybe it would be worth adding a comment in the code about it (nitpick).

odkq replied:
Commented on 157e5e4a

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            # Running inside gitlab-ci
+            local_env['FINAL_ROOT_DIR'] = '/builds/prpl-foundation/prplMesh'
+
+        # local_env['CURRENT_UID']= str(os.getuid()) + ':' + str(os.getgid())
```
```suggestion
```

odkq replied:
Removed on  a0ae26cc

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+        return return_code
+
+
+def vararg_callback(option, opt_str, value, parser):
```
This also seems to be unused.
If it is, please remove it.
(nitpick)

odkq replied:
Fixed in 9355d53f

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+                        help='Rebuild containers')
+    parser.add_argument('--shell', dest='shell', action='store_true',
+                        help='Run a shell on the bf container')
+    parser.add_argument('--comp', dest='comp', action='store_true',
```
It looks like those options are mutually exclusive.
If they are, maybe mention it in the help text and/or use `add_mutually_exclusive_group()` to make sure they are.
(nitpick)

odkq replied:
Indeed. Fixed in 0042f0ec

rmelotte commented on dctest.py:
```diff
@@ -0,0 +1,255 @@
+            services = Services()   # With new build id
+        # rc = services.dc(['up', 'boardfarm'])
+        # rc = services.dc(['run', '--service-ports', '--entrypoint',
+        #                  '/bin/bash', 'boardfarm'], interactive=True)
```
```suggestion
```

odkq replied:
Fixed in 44af58b5

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_base.py:
```diff
@@ -16,14 +17,18 @@ class CommandError(Exception):
         entry = " ".join((cmd, " ".join(args)))
         if exitstatus != 0:
-            raise CommandError("Error executing {}:\n{}".format(entry, res))
+            raise CommandError("Error executing {}".format(entry))
```
If I'm not mistaken this was reverted by mistake?
```suggestion
            raise CommandError("Error executing {}:\n{}".format(entry, res))
```

odkq replied:
Fixed on 344ebfeb

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,116 @@
+                vol = '{}:/tmp/beerocks/logs'.format(log_path)
+
+            full_args += ["run", "--rm", "-v", vol]
+            # full_args += ["--entrypoint", entrypoint + ' ' + start]
```
```suggestion
```

odkq replied:
Fixed in 28e53f28

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,116 @@
+            full_args += args
+
+        print('_docker_compose: {}'.format(' '.join(full_args)))
+        # os.environ['CURRENT_UID'] = '1000:998'
```
```suggestion
```

odkq replied:
Fixed in 28e53f28

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,116 @@
+        if os.getenv('CI_PIPELINE_ID') is None:
+            print('Setting CI_PIPELINE_ID "latest"')
+            os.environ['CI_PIPELINE_ID'] = 'latest'
+            self._run_shell_cmd("/usr/local/bin/docker-compose",
```
Do you really need the path to the docker-compose binary to be hardcoded? (if you do then you need to add a comment explaining why)
```suggestion
            self._run_shell_cmd("docker-compose",
```

(nitpick)

odkq replied:
Thought it wasn't on the path as usually root user do not have /usr/local/bin in the path. Fixed in  1df25265

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,116 @@
+        and also after test - to insure that device still operational.
+        """
+        self._run_shell_cmd("printf",
+                            ["device_get_info", "|", "nc", "-w", "1", "controller-rme", "8002"])
```
```suggestion
                            ["device_get_info", "|", "nc", "-w", "1", self.docker_name, "8002"])
```
This is my mistake, it was fixed in `prplmesh_docker.py` but not here :-)

odkq replied:
Fixed on b4adb53b

rmelotte commented on tests/boardfarm_plugins/boardfarm_prplmesh/prplmesh_config_compose.json:
```diff
@@ -0,0 +1,55 @@
+                "name": "lan2",
+                "type": "prplmesh_compose",
+                "role": "agent",
+                "delay" : 30,
```
This should have been 7 (or not there at all) I think, but it's something I did :grin:

odkq replied:
Fixed on d0e14154

rmelotte commented on tests/environment.py:
```diff
@@ -249,9 +252,9 @@ class ALEntityDocker(ALEntity):
-    # NOTE: name arg can be also extracted from the device class itself, but test_flows.py
-    # don't have it. We can remove this arg as soon, as we drop test_flows.py
-    def __init__(self, name: str, device: None = None, is_controller: bool = False):
+    def __init__(self, name: str, device: None = None, is_controller: bool = False,
```
I'm not sure why the comment was removed, test_flows.py still needs it AFAIK.

abelog replied:
This comment is still required.

odkq replied:
Fixed on c34ceebc

rmelotte commented on tests/environment.py:
```diff
@@ -268,16 +271,19 @@ def __init__(self, name: str, device: None = None, is_controller: bool = False):
-                'ip', '-f', 'inet', 'addr', 'show', self.bridge_name)
-            device_ip = re.search(r'inet (?P<ip>[0-9.]+)',
-                                  device_ip_output.decode('utf-8')).group('ip')
+            device_ip = self.device.docker_name
```
Shouldn't this work in the non-compose version as well? (maybe not on wsl, but on Linux as long as the containers are in the same docker network this should work as well I believe).
(nitpick)

odkq replied:
Would need to be tested, but as the non-compose code is going to eventually be deleted ...

rmelotte commented on tools/docker/boardfarm-ci/Dockerfile:
```diff
@@ -1,41 +1,29 @@
 
+RUN DEBIAN_FRONTEND=noninteractive apt-get install wireshark-common -y
+# VOLUME "/home/pablo/assia/prpl/git"
+# ENTRYPOINT ["/usr/local/bin/entrypoint.sh"]
```
```suggestion
```
Please don't include unused code or code that is commented out

odkq replied:
Fixed in  dd55701f

rmelotte commented on tools/docker/boardfarm-ci/Dockerfile:
```diff
@@ -1,41 +1,29 @@
 
-# TODO: what needs this?
-#RUN pip3 install jsonschema distro
+RUN apt-get install apt-transport-https ca-certificates curl gnupg-agent software-properties-common -y
```
This whole commit is not very nice, because it splits the package installation for no reason, and also runs `apt-get install` without previously running `update`, so it will break if apt's cache is cached for long enough (say if docker tries to rebuild from here in 3 month).

odkq replied:
There was a line with apt-get update above, though not very obvious

It would make sense to put all apt-get installations on one place, but the ordering on the file is as follows:

- Debian dependencies to install the python packages
- Installation of the python packages (pip3 install...)
- Debian dependencies to install the docker and docker compose binaries
- Installation of docker and docker compose
- Debian dependencies for the tests themselves (wireshark, etc)
- Installation of boardfarm

Given the layered nature of docker this allows for better iterations when adding changes. I do not think putting al the packages for apt-get in one line would yield significant performance benefits.

But the good ordering, from the less likely to change to the most, would be:

- Dependencies to install docker and docker-compose
- Installation of docker and docker-compose 
- Installation of boardfarm (from github)
- Debian dependencies to install python packages (for tests)
- Installation of the python packages (for tests)
- Debian dependencies used by the tests themselves

I have commented the file to make clearer this rationale (dd55701f)

rmelotte replied:
You missed my point about Docker's caching.
It's not about performances at all, personally here I don't care where you start rebuilding from, the image doesn't take that much time to build anyway.
Let me take a very concrete example that highlights why the current version is wrong.
- we do run `apt-get` update at the beginning and install some packages, everything is fine.
- we install a specific version of boardfarm and this layer is stored in cache: 
  ```
  RUN git clone https://github.com/mattsm/boardfarm.git \
    && cd boardfarm \
    && git checkout 100521fde1fb67536682cafecc2f91a6e2e8a6f8 \
    && python3 setup.py install
  ```

- next, we install some packages, and this is another layer stored in cache:
  ```
  RUN apt-get install gcc libsnmp-dev -y && apt-get clean
  ```

- now say that next month you want to update the boardfarm version you use, so you change the hash.
- docker is going to re-run the git clone thing, everything is fine.
- then comes the call to `apt-get install gcc ...` and this is where things go wrong.
  You reverted the removal of apt's index, so docker will use whatever index was stored from previous layers.
  `apt` tries to download the package from its index and.... it doesn't exist anymore because it was updated in debian's repositories.

This is NOT a nitpick comment. It effectively renders the image impossible to build after some time (unless `--no-cache` is used), so please fix it:
- if you want to keep the different calls to `apt-get` rather than keeping the single one that was there, fine for me it doesn't matter as much.
- every time you call `apt-get install` you HAVE TO do an `apt-get update` before.
- it's not strictly required to remove apt's index, but you never need it so there is no reason to keep it: `rm -rf /var/lib/apt/lists/*`

odkq replied:
Understood, Fixed on a9997080

rmelotte commented on tools/docker/boardfarm-ci/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+            - USER
+            - ROOT_DIR
+            - CURRENT_ID
+            - RUN_ID
+            - CI_PIPELINE_ID
```
What is the difference between RUN_ID and CI_PIPELINE_ID? 
Isn't CI_PIPELINE_ID sufficient? (nitpick)

odkq replied:
CI_PIPELINE_ID is marked as 'latest' when running locally, as locations for some files (root_dir, logs) are different when running locally. RUN_ID is the unique Id for that run (i.e, one that is incremented each time dcbuild is run)

rmelotte commented on tools/docker/boardfarm-ci/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+            - CURRENT_ID
+            - RUN_ID
+            - CI_PIPELINE_ID
+            - FINAL_ROOT_DIR
```
Correct me if I'm wrong, but I think both FINAL_ROOT_DIR and INSTALL_DIR are not used here and can be removed.

odkq replied:
INSTALL_DIR environment variable is needed in the runner entrypoint.sh (tools/docker/runner/entrypoint.sh)

It is formed from FINAL_ROOT_DIR in the entries for services "controller" and "agent", but needs to be in the boardfarm image environment, as boardfarm (and not dctest.py directly) is the one launching docker-compose for those images.

rmelotte reviewed (approved):
Lots of small comments, but I marked most of them as nitpicks because we anyway need a follow-up PR to remove the old boardfarm job among other things.

rmelotte commented:
Also I didn't mention it, but flake8 is complaining so you'll have to address that before you can merge: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/674432838

rmelotte commented:
I also didn't repeat it here (yet?) but in the future please try to describe what you do in the commit description.
In addition to [the contributing guide](https://github.com/prplfoundation/prplMesh/blob/master/CONTRIBUTING.md) you can check in the git history to see what we usually do (reviewing other pull request is also a good way to get familiar with it :-) )

odkq committed:
boardfarm-ci: Remove unused code

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Commented and reordered Dockerfile

Reordered the file with this ordering, from the less likely to change to
the most:

- Dependencies to install docker and docker-compose
- Installation of docker and docker-compose
- Installation of boardfarm (from github)
- Debian dependencies to install python packages (for tests)
- Installation of the python packages (for tests)
- Debian dependencies used by the tests themselves

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Scan device names

This change scans prplmesh_config_compose.js for device names and
creates the log directory for each of them found.

The scanning is also used to look for the previous run ID when running
locally, by looking for logs/<first device name>-<X>, and then using X+1
as the new ID.

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Fixes for PEP-8 compliance

Signed-off-by: pablo <pablo@odkq.com>

rmelotte commented:
The boardfarm tests were passing (in the sense that there were 2 failing tests and 2 passing ones, the 2 failing ones were disabled on master) with this commit:
`boardfarm-ci: Commented and reordered Dockerfile` : https://gitlab.com/prpl-foundation/prplMesh/-/jobs/680364644

Then all 4 started failing, among which initial AP config failed with `Can't find '\(WSC M2 Encrypted Settings\)'` with this commit:
`boardfarm-ci: Scan device names`: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/680577567

run-tests however has been failing since the beginning of this branch's history.

rmelotte committed:
tests: ALEntity: use name in place of device.docker_name

When working on the docker-compose version of the boardfarm tests, we
thought the ALEntity names had to be the same as the name boardfarm
uses for boardfarm to be able to access them.

It turns out that boardfarm only needs PrplMeshCompose to get its name
from the boardfarm configuration file, and the ALEntity can use
another one.

Since test_flows doesn't pass a "device" parameter when constructing
an ALEntity, it HAS to use "name" instead of going through the device
to access the docker_name.

environment.py:

  - Use the name of the ALEntityDocker and RadioDocker in instead of
accessing "device".

prplmesh_compose.py:

  - When creating the ALEntity, use docker_name as the name to make
  the different docker commands work.

This will make the ALEntity usable again regardless of whether the
tests runs with boardfarm or test_flows.py.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

abelog commented on dctest.py:
```diff
@@ -0,0 +1,252 @@
+#
+# https://www.python.org/ftp/python/3.5.0/Python-3.5.0.tgz
+#
+# Also, when calling a function look for 'New in version 3.X' where X > 5
```
Please, add license header.

odkq replied:
Added on 599fc486

abelog commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,115 @@
+        config = kwargs.get("config", kwargs)
+
+        # List of device's consoles test can interact with
+        self.consoles = [self]
```
There is no consoles created in the device constructor, so i don't really get why do we need that assignment.

odkq replied:
If it is removed, boardfarm aborts:

  File "/usr/local/lib/python3.8/site-packages/boardfarm-1.1.1-py3.8.egg/boardfarm/lib/bft_logging.py", line 256, in create_file_logs
    for console in board.consoles:
AttributeError: 'PrplMeshCompose' object has no attribute 'consoles'

abelog commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,115 @@
+        self.cleanup_cmd = config.get("cleanup_cmd", None)
+        self.conn_cmd = config.get("conn_cmd", None)
+        self.delay = config.get("delay", 7)
+        self.docker_network = "boardfarm-ci_default"
```
Are we going to use one docker network for **all** test with `prplmesh_compose` devices?

odkq replied:
If more than one bridge is used, it will be created beforehand and managed by dctest.py, as the sniffer is going to be run outside (see PR #1591 )

abelog commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,115 @@
+        self.user_id = os.getenv("SUDO_USER", os.getenv("USER", ""))
+
+        self.docker_name = "-".join((config.get("name", "prplmesh_compose"), self.unique_id))
+        self.name = config.get("name", "prplmesh_compose")
```
_Nitpick_: reorder of lines will allow just to use `self.name` without additional parse of `config`. Improves readability.
```suggestion
        self.name = config.get("name", "prplmesh_compose")
        self.docker_name = "-".join((self.name, self.unique_id))
```

Also: usage of `prplmesh_compose` as `self.name` may break the tests. Currently, in the test we are locating devices by `self.name` ([example](https://github.com/prplfoundation/prplMesh/blob/master/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py#L18)). To referrer devices by name, it should be among members of boardfarm build-in [enum](https://github.com/mattsm/boardfarm/blob/master/boardfarm/lib/DeviceManager.py#L25).
But i would suggest just to abort the test is `name` isn't set in configuration file.

odkq replied:
Fixed on 356793b1

abelog commented on tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py:
```diff
@@ -0,0 +1,115 @@
+        return True
+
+    def prprlmesh_status_check(self):
+        self.check_status()
```
`prprlmesh_status_check` is used during `InitialApConfig` test to check that agent is in `OPERATIONAL` state.
`self.check_status()` just verifies that we can connect to the prplMesh instance which doesn't guarantee that it's `OPERATIONAL`.

Suggest to execute `/etc/init.d/prplmesh status` inside image with agent/controller and `Fail` if status isn't `OPERATIONAL`.

odkq replied:
We will leave the check as it is now so the merge can happen. Later on we can branch to enhance it

abelog commented on tools/docker/boardfarm-ci/Dockerfile:
```diff
@@ -1,42 +1,31 @@
+RUN pip3 install -r requirements.txt
+
+# Installation of Debian dependencies needed by the tests
+RUN DEBIAN_FRONTEND=noninteractive apt-get install wireshark-common -y && apt-get clean
```
Shouldn't we also install `tshark` which is used for wired sniffer in `test_flows.py`?

odkq replied:
Added on the last commit to Dockerfile 
a99970805

abelog reviewed (commented):
Not approving, as `tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py::prprlmesh_status_check` should be **fixed**  before merge of PR in my opinon. It may be reason for fails mentioned in @rmelotte  [comment](https://github.com/prplfoundation/prplMesh/pull/1587#issuecomment-672057211).

odkq committed:
boardfarm-ci: Remove unused code

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Remove commented out line

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Removed more unused code

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Removed commented out code

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Removed commented-out code

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Use relative path

Use relative path on docker-compose invocation, not using
/usr/local/bin/...

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Added mutual exclusion in argv

This prevents more than one --clean, --test, --build, etc being ran at
the same time

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Group all apt-get commands together

With apt-get upgrade before

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Removed unused CURRENT_ID

Renamed CURRENT_UID to CURRENT_UID_GID as suggested and removed
CURRENT_ID wich was not used by agent/controller images

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Commented on issue PPM-208

As suggested from PR comment.

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Better error message on _run_cmd

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Fixed check_status

Pass self.docker_name instead of fixed "controller-rme" when performing
device_get_info, as suggested by PR comment.

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Set delay to the default 7 seconds

On prplmesh_compose entry of prplmesh_config_compose.json

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Restored comment on ALEntityDocker's contructor

Was removed but it is still pertinent

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Added license header to dctest.py

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Reorder lines to improve readability

Signed-off-by: pablo <pablo@odkq.com>

aciliketcap commented on tools/docker/boardfarm-ci/Dockerfile:
```diff
@@ -1,42 +1,30 @@
+&& DEBIAN_FRONTEND=noninteractive apt-get install \
+apt-transport-https ca-certificates curl gnupg-agent \
+software-properties-common curl gcc libsnmp-dev wireshark-common tshark \
+-y && apt-get clean
```
Do we still need wireshark and tshark inside the boardfarm-ci image? We already know that sniffer doesn't work when we try that way. Maybe we can keep it and I'll exclude it in the PR fixing PPM-331.

odkq replied:
Yes, I can think it is better removed on the same branch that is depending on having dumpcap installed alongside python 3.5 on the host for dctest to work (and checking the dependency -- that the executable exists -- as it is done for docker and docker-compose)

odkq committed:
boardfarm-ci: Fixed typo

Missing the n when restoring the original format string with \n ...

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Aesthetical changes to pass PEP-8

An unused include and some spacess missing for PEP-8 compliance

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Resolved merge conflict in Dockerfile

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
Merge branch 'master' into feature/PPM-30-boardfarm-docker-compose

boardfarm-ci: Merging master into PPM-30-boardfarm-docker-compose

Fixing two conflicts in tests/environment.py
Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Added missing jq

Signed-off-by: pablo <pablo@odkq.com>

odkq committed:
boardfarm-ci: Fixes for the new prplmesh/prplmesh URL

Signed-off-by: pablo <pablo@odkq.com>
